decoder/vaapidecoder_vp9.cpp

```diff
@@ -90,23 +91,50 @@ void VaapiDecoderVP9::flush(void)
 
 Decode_Status VaapiDecoderVP9::ensureContext(const Vp9FrameHdr* hdr)
 {
+    bool resetContext = false;
+    Decode_Status status = DECODE_SUCCESS;
+    DEBUG("got frame size: %d x %d", hdr->width, hdr->height);
+
     // only reset va context when there is a larger frame
     if (m_configBuffer.width < hdr->width
         || m_configBuffer.height <  hdr->height) {
+        resetContext = true;
         INFO("frame size changed, reconfig codec. orig size %d x %d, new size: %d x %d",
                 m_configBuffer.width, m_configBuffer.height, hdr->width, hdr->height);
-        Decode_Status status = VaapiDecoderBase::terminateVA();
+        
+        if (m_hasContext){
+            status = VaapiDecoderBase::terminateVA();
```

Comment:
There is no need check you have context or not. we use smart pointer. reset context it in terminateVA directly is not harmful,No matter you have context or not. I have create a pull request #177 to simply this patch. please help review it.
 
Decode_Status VaapiDecoderBase::terminateVA(void)
{
    INFO("base: terminate VA");
    m_surfacePool.reset();
    DEBUG("surface pool is reset");
    m_context.reset();
    m_display.reset();

    m_VAStarted = false;
    return DECODE_SUCCESS;
}